### PR TITLE
Implementing custom upload function

### PR DESCRIPTION
**Pull request by jerinjacob007**
New configuration option 

`uploader : {
uploadByFile(file){
return ...
} `
similar to [image tool](https://github.com/editor-js/image#providing-custom-uploading-methods)

• Readme updated

### Solves
#5

### CODE REVIEW

**Review by Solweigdev — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ Attaches Tool supports these configuration parameters:
 | Field | Type     | Description        |
 | ----- | -------- | ------------------ |
 | endpoint | `string` | **Required:** endpoint for file uploading. |
+| uploader | `{uploadByFile: function}` |Optional custom uploading methods|
```
Comment (neSpecc) [inline README.md]: Please add more detailed description of this ability to the Readme

File: src/uploader.js [hunk 2]
```diff
@@ -21,18 +21,52 @@ export default class Uploader {
...
+        beforeSend: () => onPreview(),
+        fieldName: this.config.field,
+        headers: this.config.additionalRequestHeaders || {}
+      }).then((response) => response.body);
+    }
+
+    upload.then((response) => {
       this.onUpload(response);
     }).catch((error) => {
+      console.warn(error);
```
Comment (neSpecc) [inline src/uploader.js]: do we need this log?

File: README.md (hunk 1 above)
Comment (jerinjacob007) [reply README.md]: Ok I'll do that

File: src/uploader.js (hunk 2 above)
Comment (jerinjacob007) [reply src/uploader.js]: I can remove it

**Comment by christoph-kluge** [comment]
Is this going to be merged anytime soon? What is misisng here?

**Review by neSpecc — APPROVED** [review]

**Comment by neSpecc** [comment]
Please increment a minor version and we will be ready to merge

**Comment by jerinjacob007** [comment]
> Please increment a minor version and we will be ready to merge

Version changed to `1.1.2`

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@editorjs/attaches",
-  "version": "1.1.1",
+  "version": "1.1.2",
```
Comment (neSpecc) [inline package.json]: There is the new feature added, so `1.2.0` would be better

**Comment by jerinjacob007** [comment]
Version is `1.2.0` now

**Review by christoph-kluge — APPROVED** [review]

**Pull request merged**